### PR TITLE
fixes #743 made data classes final

### LINKED ISSUE #743: Make all data classes in API final
All of the data classes in the Fluo API are intended to be immutable.   However since the classes are not final, someone could extend them to make a mutable version and pass that mutable version to Fluo. Should make all of the data classes final.  For Bytes this will require completely dropping mutable bytes and moving all of its code to Bytes.  Can also drop the internal factory interfaces in Bytes.

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

**Comment by mikewalch** [comment]
Now that Bytes objects are no longer created using the MutableBytes in core.  I think we should move back the tests in `org.apache.fluo.core.data` to `org.apache.fluo.api.data`.

File: modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java [hunk 1]
```diff
@@ -118,7 +155,7 @@ public Bytes() {}
    * @param b2 second byte sequence to compare
    * @return comparison result
    */
-  public static final int compareBytes(Bytes b1, Bytes b2) {
+  private static final int compareBytes(Bytes b1, Bytes b2) {
```
Comment (mikewalch) [inline modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java]: Given that this is now private, how about moving it's description of how the compare works to the javadoc of the public method `compareTo(Bytes other)`?

**Comment by keith-turner** [comment]
> Now that Bytes objects are no longer created using the MutableBytes in core. I think we should move back the tests in org.apache.fluo.core.data to org.apache.fluo.api.data.

Good point.  However, I think the vint thing will impede this effort.  I was thinking I would like to get rid of it too.  Maybe  I can do that in a follow on PR and move the test.

**Comment by keith-turner** [comment]
Re removing vint thing, I opened #746.

**Comment by mikewalch** [comment]
+1

**Pull request merged**